src/org/zaproxy/zap/model/ValueGenerator.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2016 The ZAP Development Team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.model;
+
+import java.net.URI;
+import java.util.Map;
+
+public interface ValueGenerator {

```

Comment:
I'm having second thoughts about this interface (and I was involved in specifying it;)
We have quite a few usecases for this enhancement - HTML forms submitted by the std and ajax spiders, WSDL defns and OpenAPI defns.
The interface seems too targeted at the HTML forms, and its really not clear what keys should be used in other cases.
I also suspect that in most cases implementations will just use the field name ;)
So how about we pull out the field name as a string but also add a type string, which could be things like "htmlform", "wsdl", "openapi" etc - it shouldnt be an enum as we'd want it to be as extensible as possible. That way implementations can target specific types of fields.
And also maybe rename formAttributes to something more generic like envAttributes?
It would also be good to document (somewhere) what the keys actually are, eg for HTML form the tag forms, and give some examples. This will be more important for the WSDL / OpenAPI attributes as it will be much less clear what they will be.